diff -ur boost_1_60_0/boost/type_traits/detail/mp_defer.hpp boost/boost/type_traits/detail/mp_defer.hpp
--- boost_1_60_0/boost/type_traits/detail/mp_defer.hpp	2015-12-13 07:49:42.000000000 -0500
+++ boost/boost/type_traits/detail/mp_defer.hpp	2016-01-05 10:26:26.000000000 -0500
@@ -12,6 +12,12 @@
 #include <boost/type_traits/integral_constant.hpp>
 #include <boost/type_traits/conditional.hpp>
 
+// This is needed to prevent build failures with Boost 1.60.0 (using Xcode 7.2
+// and the OS X 10.8 SDK)
+#ifdef check
+#undef check
+#endif
+
 namespace boost
 {
 
