diff -ur boost_1_69_0/boost/random/detail/integer_log2.hpp boost/boost/random/detail/integer_log2.hpp
--- boost_1_69_0/boost/random/detail/integer_log2.hpp	2018-12-05 14:58:22.000000000 -0500
+++ boost/boost/random/detail/integer_log2.hpp	2019-01-15 14:03:54.000000000 -0500
@@ -16,7 +16,7 @@
 
 #include <boost/config.hpp>
 #include <boost/limits.hpp>
-#include <boost/pending/integer_log2.hpp>
+#include <boost/integer/integer_log2.hpp>
 
 namespace boost {
 namespace random {
