diff -ur boost_1_65_0/boost/thread/pthread/timespec.hpp boost/boost/thread/pthread/timespec.hpp
--- boost_1_65_0/boost/thread/pthread/timespec.hpp	2017-08-19 12:49:49.000000000 -0400
+++ boost/boost/thread/pthread/timespec.hpp	2017-08-23 15:03:58.000000000 -0400
@@ -75,7 +75,7 @@
     {
       timespec ts;
 
-#if defined CLOCK_MONOTONIC && defined BOOST_THREAD_USEFIXES_TIMESPEC
+#if 0 && defined CLOCK_MONOTONIC && defined BOOST_THREAD_USEFIXES_TIMESPEC
       if ( ::clock_gettime( CLOCK_MONOTONIC, &ts ) )
       {
         ts.tv_sec = 0;
