diff -ur numpy-1.26.2/numpy/core/_add_newdocs.py numpy/numpy/core/_add_newdocs.py
--- numpy-1.26.2/numpy/core/_add_newdocs.py	1969-12-31 19:00:00
+++ numpy/numpy/core/_add_newdocs.py	2023-11-30 11:48:56
@@ -4970,36 +4970,40 @@
     See `global_state` for more information.
     """)
 
-add_newdoc('numpy.core._multiarray_tests', 'format_float_OSprintf_g',
-    """
-    format_float_OSprintf_g(val, precision)
-
-    Print a floating point scalar using the system's printf function,
-    equivalent to:
-
-        printf("%.*g", precision, val);
-
-    for half/float/double, or replacing 'g' by 'Lg' for longdouble. This
-    method is designed to help cross-validate the format_float_* methods.
-
-    Parameters
-    ----------
-    val : python float or numpy floating scalar
-        Value to format.
-
-    precision : non-negative integer, optional
-        Precision given to printf.
-
-    Returns
-    -------
-    rep : string
-        The string representation of the floating point value
-
-    See Also
-    --------
-    format_float_scientific
-    format_float_positional
-    """)
+#
+# Don't try to add docs to numpy.core._multiarray_tests, since we don't
+# package that
+#
+#add_newdoc('numpy.core._multiarray_tests', 'format_float_OSprintf_g',
+#    """
+#    format_float_OSprintf_g(val, precision)
+#
+#    Print a floating point scalar using the system's printf function,
+#    equivalent to:
+#
+#        printf("%.*g", precision, val);
+#
+#    for half/float/double, or replacing 'g' by 'Lg' for longdouble. This
+#    method is designed to help cross-validate the format_float_* methods.
+#
+#    Parameters
+#    ----------
+#    val : python float or numpy floating scalar
+#        Value to format.
+#
+#    precision : non-negative integer, optional
+#        Precision given to printf.
+#
+#    Returns
+#    -------
+#    rep : string
+#        The string representation of the floating point value
+#
+#    See Also
+#    --------
+#    format_float_scientific
+#    format_float_positional
+#    """)
 
 
 ##############################################################################
diff -ur numpy-1.26.2/vendored-meson/meson/mesonbuild/linkers/linkers.py numpy/vendored-meson/meson/mesonbuild/linkers/linkers.py
--- numpy-1.26.2/vendored-meson/meson/mesonbuild/linkers/linkers.py	1969-12-31 19:00:00
+++ numpy/vendored-meson/meson/mesonbuild/linkers/linkers.py	2023-11-30 11:48:21
@@ -767,7 +767,7 @@
         return self._apply_prefix('-undefined,dynamic_lookup')
 
     def get_std_shared_module_args(self, options: 'KeyedOptionDictType') -> T.List[str]:
-        return ['-bundle'] + self._apply_prefix('-undefined,dynamic_lookup')
+        return self._apply_prefix('-dylib') + self._apply_prefix('-make_mergeable') + self._apply_prefix('-undefined,dynamic_lookup')
 
     def get_pie_args(self) -> T.List[str]:
         return []
