diff -ur numpy-1.24.2/numpy/core/setup.py numpy/numpy/core/setup.py
--- numpy-1.24.2/numpy/core/setup.py	2023-02-05 12:13:47
+++ numpy/numpy/core/setup.py	2023-04-13 11:34:00
@@ -736,6 +736,7 @@
         ):
             is_cpp = lang == 'c++'
             if is_cpp:
+                continue
                 # this a workaround to get rid of invalid c++ flags
                 # without doing big changes to config.
                 # c tested first, compiler should be here
