diff -ur numpy-1.26.2/numpy/core/src/multiarray/_multiarray_tests.c.src numpy/numpy/core/src/multiarray/_multiarray_tests.c.src
--- numpy-1.26.2/numpy/core/src/multiarray/_multiarray_tests.c.src	1969-12-31 19:00:00
+++ numpy/numpy/core/src/multiarray/_multiarray_tests.c.src	2023-11-20 17:34:44
@@ -63,6 +63,7 @@
 
 
 /** Function to test calling via ctypes */
+Py_EXPORTED_SYMBOL
 EXPORT(void*) forward_pointer(void *x)
 {
     return x;
diff -ur numpy-1.26.2/numpy/core/tests/test_limited_api.py numpy/numpy/core/tests/test_limited_api.py
--- numpy-1.26.2/numpy/core/tests/test_limited_api.py	1969-12-31 19:00:00
+++ numpy/numpy/core/tests/test_limited_api.py	2023-11-20 17:34:59
@@ -8,6 +8,7 @@
 from numpy.testing import IS_WASM
 
 
+@pytest.mark.skip('Not relevant')
 @pytest.mark.skipif(IS_WASM, reason="Can't start subprocess")
 @pytest.mark.xfail(
     sysconfig.get_config_var("Py_DEBUG"),
diff -ur numpy-1.26.2/numpy/core/tests/test_mem_policy.py numpy/numpy/core/tests/test_mem_policy.py
--- numpy-1.26.2/numpy/core/tests/test_mem_policy.py	1969-12-31 19:00:00
+++ numpy/numpy/core/tests/test_mem_policy.py	2023-11-20 17:34:59
@@ -14,6 +14,9 @@
 # these tests.
 
 
+pytestmark = pytest.mark.skip('Not relevant')
+
+
 @pytest.fixture
 def get_module(tmp_path):
     """ Add a memory policy that returns a false pointer 64 bytes into the
diff -ur numpy-1.26.2/numpy/tests/test_reloading.py numpy/numpy/tests/test_reloading.py
--- numpy-1.26.2/numpy/tests/test_reloading.py	1969-12-31 19:00:00
+++ numpy/numpy/tests/test_reloading.py	2023-11-20 17:34:59
@@ -44,6 +44,7 @@
                                           protocol=proto)) is np._NoValue)
 
 
+@pytest.mark.skip('Not relevant')
 @pytest.mark.skipif(IS_WASM, reason="can't start subprocess")
 def test_full_reimport():
     """At the time of writing this, it is *not* truly supported, but
