diff -ur numpy-1.17.0/numpy/core/_dtype_ctypes.py numpy/numpy/core/_dtype_ctypes.py
--- numpy-1.17.0/numpy/core/_dtype_ctypes.py	2019-07-26 10:10:50.000000000 -0400
+++ numpy/numpy/core/_dtype_ctypes.py	2019-08-12 10:43:29.000000000 -0400
@@ -22,8 +22,8 @@
 * PEP3118 cannot represent unions, but both numpy and ctypes can
 * ctypes cannot handle big-endian structs with PEP3118 (bpo-32780)
 """
-import _ctypes
-import ctypes
+#import _ctypes
+#import ctypes
 
 import numpy as np
 
