diff -ur numpy-1.26.2/vendored-meson/meson/mesonbuild/linkers/linkers.py numpy/vendored-meson/meson/mesonbuild/linkers/linkers.py
--- numpy-1.26.2/vendored-meson/meson/mesonbuild/linkers/linkers.py	1969-12-31 19:00:00
+++ numpy/vendored-meson/meson/mesonbuild/linkers/linkers.py	2023-11-30 11:48:21
@@ -767,7 +767,7 @@
         return self._apply_prefix('-undefined,dynamic_lookup')
 
     def get_std_shared_module_args(self, options: 'KeyedOptionDictType') -> T.List[str]:
-        return ['-bundle'] + self._apply_prefix('-undefined,dynamic_lookup')
+        return self._apply_prefix('-dylib') + self._apply_prefix('-make_mergeable') + self._apply_prefix('-undefined,dynamic_lookup')
 
     def get_pie_args(self) -> T.List[str]:
         return []
