diff -ur numpy-1.15.0/numpy/distutils/system_info.py numpy/numpy/distutils/system_info.py
--- numpy-1.15.0/numpy/distutils/system_info.py	2018-07-23 08:27:00.000000000 -0400
+++ numpy/numpy/distutils/system_info.py	2018-08-09 10:20:28.000000000 -0400
@@ -1892,7 +1892,8 @@
         libraries = [lib.strip().lower() for lib in libraries]
 
         if (sys.platform == 'darwin' and
-                not os.getenv('_PYTHON_HOST_PLATFORM', None)):
+                #not os.getenv('_PYTHON_HOST_PLATFORM', None)):
+                True):
             # Use the system BLAS from Accelerate or vecLib under OSX
             args = []
             link_args = []
