diff -ur numpy-1.24.2/numpy/core/tests/test_limited_api.py numpy/numpy/core/tests/test_limited_api.py
--- numpy-1.24.2/numpy/core/tests/test_limited_api.py	2023-02-05 12:13:47
+++ numpy/numpy/core/tests/test_limited_api.py	2023-04-13 11:35:38
@@ -8,6 +8,7 @@
 from numpy.testing import IS_WASM
 
 
+@pytest.mark.skip('Not relevant')
 @pytest.mark.skipif(IS_WASM, reason="Can't start subprocess")
 @pytest.mark.xfail(
     sysconfig.get_config_var("Py_DEBUG"),
diff -ur numpy-1.24.2/numpy/core/tests/test_mem_policy.py numpy/numpy/core/tests/test_mem_policy.py
--- numpy-1.24.2/numpy/core/tests/test_mem_policy.py	2023-02-05 12:13:47
+++ numpy/numpy/core/tests/test_mem_policy.py	2023-04-13 11:37:15
@@ -9,6 +9,9 @@
 import sys
 
 
+pytestmark = pytest.mark.skip('Not relevant')
+
+
 @pytest.fixture
 def get_module(tmp_path):
     """ Add a memory policy that returns a false pointer 64 bytes into the
diff -ur numpy-1.24.2/numpy/tests/test_reloading.py numpy/numpy/tests/test_reloading.py
--- numpy-1.24.2/numpy/tests/test_reloading.py	2023-02-05 12:13:47
+++ numpy/numpy/tests/test_reloading.py	2023-04-13 11:37:56
@@ -44,6 +44,7 @@
                                           protocol=proto)) is np._NoValue)
 
 
+@pytest.mark.skip('Not relevant')
 @pytest.mark.skipif(IS_WASM, reason="can't start subprocess")
 def test_full_reimport():
     """At the time of writing this, it is *not* truly supported, but
