diff -ur numpy-1.22.2/numpy/distutils/cpuinfo.py numpy/numpy/distutils/cpuinfo.py
--- numpy-1.22.2/numpy/distutils/cpuinfo.py	2022-01-10 12:36:38.000000000 -0500
+++ numpy/numpy/distutils/cpuinfo.py	2022-02-22 12:11:48.000000000 -0500
@@ -25,6 +25,8 @@
 
 
 def getoutput(cmd, successful_status=(0,), stacklevel=1):
+    # Can't open a subprocess on iOS
+    return False, ""
     try:
         status, output = getstatusoutput(cmd)
     except OSError as e:
