diff -ur numpy-1.23.1/numpy/core/setup.py numpy/numpy/core/setup.py
--- numpy-1.23.1/numpy/core/setup.py	2022-07-08 19:01:04.000000000 -0400
+++ numpy/numpy/core/setup.py	2022-07-25 14:09:41.000000000 -0400
@@ -739,6 +739,7 @@
         ):
             is_cpp = lang == 'c++'
             if is_cpp:
+                continue
                 # this a workaround to get rid of invalid c++ flags
                 # without doing big changes to config.
                 # c tested first, compiler should be here
