diff -ur numpy-2.1.3/numpy/_core/src/multiarray/_multiarray_tests.c.src numpy/numpy/_core/src/multiarray/_multiarray_tests.c.src
--- numpy-2.1.3/numpy/_core/src/multiarray/_multiarray_tests.c.src	2024-11-02 11:21:28
+++ numpy/numpy/_core/src/multiarray/_multiarray_tests.c.src	2024-11-05 14:54:45
@@ -84,6 +84,7 @@
 
 
 /** Function to test calling via ctypes */
+Py_EXPORTED_SYMBOL
 EXPORT(void*) forward_pointer(void *x)
 {
     return x;
diff -ur numpy-2.1.3/numpy/_core/tests/test_array_interface.py numpy/numpy/_core/tests/test_array_interface.py
--- numpy-2.1.3/numpy/_core/tests/test_array_interface.py	2024-11-02 11:21:28
+++ numpy/numpy/_core/tests/test_array_interface.py	2024-11-05 14:56:09
@@ -9,6 +9,7 @@
     """ Some codes to generate data and manage temporary buffers use when
     sharing with numpy via the array interface protocol.
     """
+    pytest.skip('Not relevant')
     if sys.platform.startswith('cygwin'):
         pytest.skip('link fails on cygwin')
     if IS_WASM:
diff -ur numpy-2.1.3/numpy/_core/tests/test_limited_api.py numpy/numpy/_core/tests/test_limited_api.py
--- numpy-2.1.3/numpy/_core/tests/test_limited_api.py	2024-11-02 11:21:28
+++ numpy/numpy/_core/tests/test_limited_api.py	2024-11-05 14:54:45
@@ -68,6 +68,7 @@
 
 
 
+@pytest.mark.skip('Not relevant')
 @pytest.mark.skipif(IS_WASM, reason="Can't start subprocess")
 @pytest.mark.xfail(
     sysconfig.get_config_var("Py_DEBUG"),
diff -ur numpy-2.1.3/numpy/_core/tests/test_mem_policy.py numpy/numpy/_core/tests/test_mem_policy.py
--- numpy-2.1.3/numpy/_core/tests/test_mem_policy.py	2024-11-02 11:21:28
+++ numpy/numpy/_core/tests/test_mem_policy.py	2024-11-05 14:54:45
@@ -12,6 +12,9 @@
 from numpy._core.multiarray import get_handler_name
 
 
+pytestmark = pytest.mark.skip('Not relevant')
+
+
 @pytest.fixture
 def get_module(tmp_path):
     """ Add a memory policy that returns a false pointer 64 bytes into the
diff -ur numpy-2.1.3/numpy/f2py/tests/util.py numpy/numpy/f2py/tests/util.py
--- numpy-2.1.3/numpy/f2py/tests/util.py	2024-11-02 11:21:28
+++ numpy/numpy/f2py/tests/util.py	2024-11-05 15:00:06
@@ -54,7 +54,7 @@
                 cwd=tmpdir,
                 capture_output=True,
             )
-        except subprocess.CalledProcessError:
+        except (subprocess.CalledProcessError, FileNotFoundError):
             pytest.skip("meson not present, skipping compiler dependent test", allow_module_level=True)
         return runmeson.returncode == 0
     finally:
diff -ur numpy-2.1.3/numpy/tests/test_configtool.py numpy/numpy/tests/test_configtool.py
--- numpy-2.1.3/numpy/tests/test_configtool.py	2024-11-02 11:21:28
+++ numpy/numpy/tests/test_configtool.py	2024-11-05 14:56:56
@@ -8,6 +8,8 @@
 from numpy.testing import IS_WASM
 
 
+pytestmark = pytest.mark.skip('Not relevant')
+
 is_editable = not bool(np.__path__)
 numpy_in_sitepackages = sysconfig.get_path('platlib') in np.__file__
 # We only expect to have a `numpy-config` available if NumPy was installed via
diff -ur numpy-2.1.3/numpy/tests/test_reloading.py numpy/numpy/tests/test_reloading.py
--- numpy-2.1.3/numpy/tests/test_reloading.py	2024-11-02 11:21:28
+++ numpy/numpy/tests/test_reloading.py	2024-11-05 14:54:45
@@ -46,6 +46,7 @@
                                           protocol=proto)) is np._NoValue)
 
 
+@pytest.mark.skip('Not relevant')
 @pytest.mark.skipif(IS_WASM, reason="can't start subprocess")
 def test_full_reimport():
     """At the time of writing this, it is *not* truly supported, but
