diff -ur Python-3.9.5/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.9.5/pyconfig.h.in	2021-05-03 10:54:42.000000000 -0400
+++ Python/pyconfig.h.in	2021-06-25 16:02:52.000000000 -0400
@@ -773,7 +773,7 @@
 #undef HAVE_PREAD
 
 /* Define to 1 if you have the `preadv' function. */
-#undef HAVE_PREADV
+/* #undef HAVE_PREADV */
 
 /* Define to 1 if you have the `preadv2' function. */
 #undef HAVE_PREADV2
@@ -815,7 +815,7 @@
 #undef HAVE_PWRITE
 
 /* Define to 1 if you have the `pwritev' function. */
-#undef HAVE_PWRITEV
+/* #undef HAVE_PWRITEV */
 
 /* Define to 1 if you have the `pwritev2' function. */
 #undef HAVE_PWRITEV2
