diff -ur Python-3.11.3/Lib/_osx_support.py Python/Lib/_osx_support.py
--- Python-3.11.3/Lib/_osx_support.py	2023-04-04 18:22:17
+++ Python/Lib/_osx_support.py	2023-04-14 10:44:10
@@ -385,11 +385,11 @@
             except ValueError:
                 break
 
-    elif not _supports_arm64_builds():
-        # Look for "-arch arm64" and drop that
-        for idx in reversed(range(len(compiler_so))):
-            if compiler_so[idx] == '-arch' and compiler_so[idx+1] == "arm64":
-                del compiler_so[idx:idx+2]
+    #elif not _supports_arm64_builds():
+    #    # Look for "-arch arm64" and drop that
+    #    for idx in reversed(range(len(compiler_so))):
+    #        if compiler_so[idx] == '-arch' and compiler_so[idx+1] == "arm64":
+    #            del compiler_so[idx:idx+2]
 
     if 'ARCHFLAGS' in os.environ and not stripArch:
         # User specified different -arch flags in the environ,
diff -ur Python-3.11.3/Makefile.pre.in Python/Makefile.pre.in
--- Python-3.11.3/Makefile.pre.in	2023-04-04 18:22:17
+++ Python/Makefile.pre.in	2023-04-14 10:44:50
@@ -2086,30 +2086,30 @@
 	fi
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST)  $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -Wi $(DESTDIR)$(LIBDEST)/compileall.py \
-		-j0 -d $(LIBDEST) -f \
+		-j1 -d $(LIBDEST) -f \
 		-x 'bad_coding|badsyntax|site-packages|lib2to3/tests/data' \
 		$(DESTDIR)$(LIBDEST)
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -Wi -O $(DESTDIR)$(LIBDEST)/compileall.py \
-		-j0 -d $(LIBDEST) -f \
+		-j1 -d $(LIBDEST) -f \
 		-x 'bad_coding|badsyntax|site-packages|lib2to3/tests/data' \
 		$(DESTDIR)$(LIBDEST)
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -Wi -OO $(DESTDIR)$(LIBDEST)/compileall.py \
-		-j0 -d $(LIBDEST) -f \
+		-j1 -d $(LIBDEST) -f \
 		-x 'bad_coding|badsyntax|site-packages|lib2to3/tests/data' \
 		$(DESTDIR)$(LIBDEST)
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -Wi $(DESTDIR)$(LIBDEST)/compileall.py \
-		-j0 -d $(LIBDEST)/site-packages -f \
+		-j1 -d $(LIBDEST)/site-packages -f \
 		-x badsyntax $(DESTDIR)$(LIBDEST)/site-packages
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -Wi -O $(DESTDIR)$(LIBDEST)/compileall.py \
-		-j0 -d $(LIBDEST)/site-packages -f \
+		-j1 -d $(LIBDEST)/site-packages -f \
 		-x badsyntax $(DESTDIR)$(LIBDEST)/site-packages
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -Wi -OO $(DESTDIR)$(LIBDEST)/compileall.py \
-		-j0 -d $(LIBDEST)/site-packages -f \
+		-j1 -d $(LIBDEST)/site-packages -f \
 		-x badsyntax $(DESTDIR)$(LIBDEST)/site-packages
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -m lib2to3.pgen2.driver $(DESTDIR)$(LIBDEST)/lib2to3/Grammar.txt
diff -ur Python-3.11.3/Modules/Setup Python/Modules/Setup
--- Python-3.11.3/Modules/Setup	2023-04-04 18:22:17
+++ Python/Modules/Setup	2023-04-14 10:42:39
@@ -146,7 +146,7 @@
 #_random _randommodule.c
 #_socket socketmodule.c
 #_statistics _statisticsmodule.c
-#_struct _struct.c
+_struct _struct.c
 #_typing _typingmodule.c
 #_zoneinfo _zoneinfo.c
 #array arraymodule.c
@@ -177,7 +177,7 @@
 #_codecs_kr cjkcodecs/_codecs_kr.c
 #_codecs_tw cjkcodecs/_codecs_tw.c
 #_multibytecodec cjkcodecs/multibytecodec.c
-#unicodedata unicodedata.c
+unicodedata unicodedata.c
 
 # Modules with some UNIX dependencies
 
diff -ur Python-3.11.3/configure Python/configure
--- Python-3.11.3/configure	2023-04-04 18:22:17
+++ Python/configure	2023-04-14 10:37:54
@@ -3408,7 +3408,7 @@
 
 
 
-define_xopen_source=yes
+define_xopen_source=no
 
 # Arguments passed to configure.
 
@@ -3832,6 +3832,9 @@
 	*-*-wasi)
 	    ac_sys_system=WASI
 	    ;;
+	*-*-darwin)
+	    ac_sys_system=Darwin
+	    ;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -3885,6 +3888,9 @@
 	wasm32-*-* | wasm64-*-*)
 		_host_cpu=$host_cpu
 		;;
+	*-*-darwin*)
+		_host_cpu=$host_cpu
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -7209,13 +7215,13 @@
   fi
 fi
 
-if test "$cross_compiling" = yes; then
-    case "$READELF" in
-	readelf|:)
-	as_fn_error $? "readelf for the host is required for cross builds" "$LINENO" 5
-	;;
-    esac
-fi
+#if test "$cross_compiling" = yes; then
+#    case "$READELF" in
+#	readelf|:)
+#	as_fn_error $? "readelf for the host is required for cross builds" "$LINENO" 5
+#	;;
+#    esac
+#fi
 
 
 
@@ -10607,7 +10613,7 @@
             LIBTOOL_CRUFT=""
     fi
     if test "$cross_compiling" = yes; then :
-  ac_osx_32bit=yes
+  ac_osx_32bit=no
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
diff -ur Python-3.11.3/setup.py Python/setup.py
--- Python-3.11.3/setup.py	2023-04-04 18:22:17
+++ Python/setup.py	2023-04-14 10:37:54
@@ -69,6 +69,8 @@
 def get_platform():
     # Cross compiling
     if "_PYTHON_HOST_PLATFORM" in os.environ:
+        if os.environ["_PYTHON_HOST_PLATFORM"].startswith('darwin'):
+            return 'darwin'
         return os.environ["_PYTHON_HOST_PLATFORM"]
 
     # Get value of sys.platform
@@ -166,7 +168,10 @@
     for var_name in make_vars:
         var = sysconfig.get_config_var(var_name)
         if var is not None:
-            m = re.search(r'--sysroot=([^"]\S*|"[^"]+")', var)
+            if MACOS:
+                m = re.search(r'-isysroot\s+(\S+)', var)
+            else:
+                m = re.search(r'--sysroot=([^"]\S*|"[^"]+")', var)
             if m is not None:
                 sysroot = m.group(1).strip('"')
                 for subdir in subdirs:
