diff -ur Python-3.12.3/Lib/test/test_cppext/__init__.py Python/Lib/test/test_cppext/__init__.py
--- Python-3.12.3/Lib/test/test_cppext/__init__.py	2024-04-09 04:09:14
+++ Python/Lib/test/test_cppext/__init__.py	2024-05-06 15:24:46
@@ -14,6 +14,7 @@
 SETUP = os.path.join(os.path.dirname(__file__), 'setup.py')
 
 
+@unittest.skip('Not relevant')
 @support.requires_subprocess()
 class TestCPPExt(unittest.TestCase):
     @support.requires_resource('cpu')
