diff -ur Python-3.12.3/configure Python/configure
--- Python-3.12.3/configure	2024-04-09 04:09:14
+++ Python/configure	2024-05-06 10:45:41
@@ -12784,8 +12784,8 @@
 			as_fn_error $? "MACOSX_DEPLOYMENT_TARGET too old ($MACOSX_DEPLOYMENT_TARGET), only 10.3 or later is supported" "$LINENO" 5
 		else
 			# building for OS X 10.3 and later
-			LDSHARED='$(CC) -bundle -undefined dynamic_lookup'
-			LDCXXSHARED='$(CXX) -bundle -undefined dynamic_lookup'
+			LDSHARED='$(CC) -Wl,-dylib -Wl,-make_mergeable -Wl,-undefined,dynamic_lookup'
+			LDCXXSHARED='$(CXX) -Wl,-dylib -Wl,-make_mergeable -Wl,-undefined,dynamic_lookup'
 			BLDSHARED="$LDSHARED"
 		fi
 		;;
