diff -ur Python-3.7.2/Lib/urllib/request.py Python/Lib/urllib/request.py
--- Python-3.7.2/Lib/urllib/request.py	2018-12-23 16:37:36.000000000 -0500
+++ Python/Lib/urllib/request.py	2019-01-15 12:53:25.000000000 -0500
@@ -2579,7 +2579,7 @@
     return False
 
 
-if sys.platform == 'darwin':
+if False and sys.platform == 'darwin':
     from _scproxy import _get_proxy_settings, _get_proxies
 
     def proxy_bypass_macosx_sysconf(host):
diff -ur Python-3.7.2/Modules/Setup.dist Python/Modules/Setup.dist
--- Python-3.7.2/Modules/Setup.dist	2018-12-23 16:37:36.000000000 -0500
+++ Python/Modules/Setup.dist	2019-01-15 12:53:51.000000000 -0500
@@ -367,3 +367,7 @@
 #
 #_sqlite3 _tkinter _curses pyexpat
 #_codecs_jp _codecs_kr _codecs_tw unicodedata
+
+*disabled*
+
+_ctypes _ctypes_test _curses _curses_panel _lzma _scproxy readline
