diff -ur Python-3.12.7/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.12.7/pyconfig.h.in	2024-09-30 22:05:46
+++ Python/pyconfig.h.in	2026-09-16 08:17:25
@@ -297,7 +297,7 @@
 #undef HAVE_DUP2
 
 /* Define to 1 if you have the `dup3' function. */
-#undef HAVE_DUP3
+/* #undef HAVE_DUP3 */
 
 /* Define if you have the '_dyld_shared_cache_contains_path' function. */
 #undef HAVE_DYLD_SHARED_CACHE_CONTAINS_PATH
@@ -885,7 +885,7 @@
 #undef HAVE_PIPE
 
 /* Define to 1 if you have the `pipe2' function. */
-#undef HAVE_PIPE2
+/* #undef HAVE_PIPE2 */
 
 /* Define to 1 if you have the `plock' function. */
 #undef HAVE_PLOCK
