diff -ur Python-3.11.3/Modules/Setup Python/Modules/Setup
--- Python-3.11.3/Modules/Setup	2023-04-04 18:22:17
+++ Python/Modules/Setup	2023-04-14 09:40:37
@@ -205,7 +205,7 @@
 #_gdbm _gdbmmodule.c -lgdbm
 #_lzma _lzmamodule.c -llzma
 #_uuid _uuidmodule.c -luuid
-#zlib  zlibmodule.c -lz
+zlib  zlibmodule.c -lz
 
 # The readline module also supports libeditline (-leditline).
 # Some systems may require -ltermcap or -ltermlib.
