diff -ur Python-3.7.0/Modules/timemodule.c Python/Modules/timemodule.c
--- Python-3.7.0/Modules/timemodule.c	2018-06-26 23:07:35.000000000 -0400
+++ Python/Modules/timemodule.c	2018-07-06 12:45:57.000000000 -0400
@@ -1693,6 +1693,7 @@
     /* Set, or reset, module variables like time.timezone */
     PyInit_timezone(m);
 
+/*
 #ifdef CLOCK_REALTIME
     PyModule_AddIntMacro(m, CLOCK_REALTIME);
 #endif
@@ -1702,15 +1703,18 @@
 #ifdef CLOCK_MONOTONIC_RAW
     PyModule_AddIntMacro(m, CLOCK_MONOTONIC_RAW);
 #endif
+*/
 #ifdef CLOCK_HIGHRES
     PyModule_AddIntMacro(m, CLOCK_HIGHRES);
 #endif
+/*
 #ifdef CLOCK_PROCESS_CPUTIME_ID
     PyModule_AddIntMacro(m, CLOCK_PROCESS_CPUTIME_ID);
 #endif
 #ifdef CLOCK_THREAD_CPUTIME_ID
     PyModule_AddIntMacro(m, CLOCK_THREAD_CPUTIME_ID);
 #endif
+*/
 #ifdef CLOCK_PROF
     PyModule_AddIntMacro(m, CLOCK_PROF);
 #endif
diff -ur Python-3.7.0/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.7.0/pyconfig.h.in	2018-07-06 12:44:24.000000000 -0400
+++ Python/pyconfig.h.in	2018-07-06 12:47:00.000000000 -0400
@@ -120,13 +120,13 @@
 #undef HAVE_CLOCK
 
 /* Define to 1 if you have the `clock_getres' function. */
-#undef HAVE_CLOCK_GETRES
+/* #undef HAVE_CLOCK_GETRES */
 
 /* Define to 1 if you have the `clock_gettime' function. */
-#undef HAVE_CLOCK_GETTIME
+/* #undef HAVE_CLOCK_GETTIME */
 
 /* Define to 1 if you have the `clock_settime' function. */
-#undef HAVE_CLOCK_SETTIME
+/* #undef HAVE_CLOCK_SETTIME */
 
 /* Define if the C compiler supports computed gotos. */
 #undef HAVE_COMPUTED_GOTOS
@@ -401,7 +401,7 @@
 #undef HAVE_GETC_UNLOCKED
 
 /* Define to 1 if you have the `getentropy' function. */
-#undef HAVE_GETENTROPY
+/* #undef HAVE_GETENTROPY */
 
 /* Define to 1 if you have the `getgrouplist' function. */
 #undef HAVE_GETGROUPLIST
