diff -ur Python-3.8.5/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.8.5/pyconfig.h.in	2020-07-20 09:01:32.000000000 -0400
+++ Python/pyconfig.h.in	2020-09-04 16:29:19.000000000 -0400
@@ -424,7 +424,7 @@
 #undef HAVE_GETC_UNLOCKED
 
 /* Define to 1 if you have the `getentropy' function. */
-#undef HAVE_GETENTROPY
+/* #undef HAVE_GETENTROPY */
 
 /* Define to 1 if you have the `getgrgid_r' function. */
 #undef HAVE_GETGRGID_R
