diff -ur Python-3.12.3/configure Python/configure
--- Python-3.12.3/configure	2024-04-09 04:09:14
+++ Python/configure	2024-05-06 10:39:26
@@ -3842,7 +3842,7 @@
 
 
 
-define_xopen_source=yes
+define_xopen_source=no
 
 # Arguments passed to configure.
 
@@ -4279,6 +4279,9 @@
 	*-*-wasi)
 	    ac_sys_system=WASI
 	    ;;
+	*-*-darwin)
+	    ac_sys_system=Darwin
+	    ;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -4339,6 +4342,9 @@
 		_host_cpu=$host_cpu
 		;;
 	wasm32-*-* | wasm64-*-*)
+		_host_cpu=$host_cpu
+		;;
+	*-*-darwin*)
 		_host_cpu=$host_cpu
 		;;
 	*)
