diff -ur Python-3.9.5/Modules/posixmodule.c Python/Modules/posixmodule.c
--- Python-3.9.5/Modules/posixmodule.c	2021-05-03 10:54:42.000000000 -0400
+++ Python/Modules/posixmodule.c	2021-06-28 12:07:02.000000000 -0400
@@ -327,7 +327,7 @@
 #    define HAVE_KILL       1
 #    define HAVE_OPENDIR    1
 #    define HAVE_PIPE       1
-#    define HAVE_SYSTEM     1
+/* #    define HAVE_SYSTEM     1 */ /* system is prohibited on iOS */
 #    define HAVE_WAIT       1
 #    define HAVE_TTYNAME    1
 #  endif  /* _MSC_VER */
diff -ur Python-3.9.5/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.9.5/pyconfig.h.in	2021-05-03 10:54:42.000000000 -0400
+++ Python/pyconfig.h.in	2021-06-28 12:07:02.000000000 -0400
@@ -134,7 +134,7 @@
 #undef HAVE_CLOCK_GETTIME
 
 /* Define to 1 if you have the `clock_settime' function. */
-#undef HAVE_CLOCK_SETTIME
+/* #undef HAVE_CLOCK_SETTIME */ /* clock_settime is prohibited on iOS */
 
 /* Define if the C compiler supports computed gotos. */
 #undef HAVE_COMPUTED_GOTOS
