diff -ur Python-3.11.3/setup.py Python/setup.py
--- Python-3.11.3/setup.py	2023-04-04 18:22:17
+++ Python/setup.py	2023-04-14 09:44:21
@@ -1420,8 +1420,12 @@
         # function my_sqrt() needs libm for sqrt()
         self.addext(Extension('_ctypes_test', ['_ctypes/_ctypes_test.c']))
 
-        ffi_inc = sysconfig.get_config_var("LIBFFI_INCLUDEDIR")
-        ffi_lib = None
+        #ffi_inc = sysconfig.get_config_var("LIBFFI_INCLUDEDIR")
+        #ffi_lib = None
+        # Use our libffi
+        ffi_inc = sysconfig.get_config_var('INCLUDEDIR')
+        ext.library_dirs.append(sysconfig.get_config_var('LIBDIR'))
+        ffi_lib = 'ffi'
 
         ffi_inc_dirs = self.inc_dirs.copy()
         if MACOS:
