diff -ur Python-3.10.2/Lib/ctypes/test/test_macholib.py Python/Lib/ctypes/test/test_macholib.py
--- Python-3.10.2/Lib/ctypes/test/test_macholib.py	2022-01-13 13:52:14.000000000 -0500
+++ Python/Lib/ctypes/test/test_macholib.py	2022-02-18 16:01:55.000000000 -0500
@@ -43,6 +43,7 @@
     raise ValueError("%s not found" % (name,))
 
 class MachOTest(unittest.TestCase):
+    @unittest.skip('requires macOS 11 or later')
     @unittest.skipUnless(sys.platform == "darwin", 'OSX-specific test')
     def test_find(self):
         # On Mac OS 11, system dylibs are only present in the shared cache,
diff -ur Python-3.10.2/Modules/_ctypes/callproc.c Python/Modules/_ctypes/callproc.c
--- Python-3.10.2/Modules/_ctypes/callproc.c	2022-01-13 13:52:14.000000000 -0500
+++ Python/Modules/_ctypes/callproc.c	2022-02-18 15:59:59.000000000 -0500
@@ -1447,6 +1447,7 @@
 #define HAVE_DYLD_SHARED_CACHE_CONTAINS_PATH_RUNTIME \
     __builtin_available(macOS 11.0, iOS 14.0, tvOS 14.0, watchOS 7.0, *)
 #else
+/*
 // Support the deprecated case of compiling on an older macOS version
 static void *libsystem_b_handle;
 static bool (*_dyld_shared_cache_contains_path)(const char *path);
@@ -1463,6 +1464,7 @@
         dlclose(libsystem_b_handle);
     }
 }
+*/
 #define HAVE_DYLD_SHARED_CACHE_CONTAINS_PATH_RUNTIME \
     _dyld_shared_cache_contains_path != NULL
 #endif
@@ -1472,6 +1474,7 @@
      PyObject *name, *name2;
      char *name_str;
 
+     /*
      if (HAVE_DYLD_SHARED_CACHE_CONTAINS_PATH_RUNTIME) {
          int r;
 
@@ -1495,9 +1498,12 @@
          }
 
      } else {
+     */
          PyErr_SetString(PyExc_NotImplementedError, "_dyld_shared_cache_contains_path symbol is missing");
          return NULL;
+     /*
      }
+     */
 
  }
 #endif
diff -ur Python-3.10.2/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.10.2/pyconfig.h.in	2022-01-13 13:52:14.000000000 -0500
+++ Python/pyconfig.h.in	2022-02-18 15:59:59.000000000 -0500
@@ -795,7 +795,7 @@
 #undef HAVE_PREAD
 
 /* Define to 1 if you have the `preadv' function. */
-#undef HAVE_PREADV
+/* #undef HAVE_PREADV */
 
 /* Define to 1 if you have the `preadv2' function. */
 #undef HAVE_PREADV2
@@ -837,7 +837,7 @@
 #undef HAVE_PWRITE
 
 /* Define to 1 if you have the `pwritev' function. */
-#undef HAVE_PWRITEV
+/* #undef HAVE_PWRITEV */
 
 /* Define to 1 if you have the `pwritev2' function. */
 #undef HAVE_PWRITEV2
