diff -ur Python-3.11.3/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.11.3/pyconfig.h.in	2023-04-04 18:22:17
+++ Python/pyconfig.h.in	2023-04-14 09:47:36
@@ -794,13 +794,13 @@
 #undef HAVE_MKFIFO
 
 /* Define to 1 if you have the `mkfifoat' function. */
-#undef HAVE_MKFIFOAT
+/* #undef HAVE_MKFIFOAT */
 
 /* Define to 1 if you have the `mknod' function. */
 #undef HAVE_MKNOD
 
 /* Define to 1 if you have the `mknodat' function. */
-#undef HAVE_MKNODAT
+/* #undef HAVE_MKNODAT */
 
 /* Define to 1 if you have the `mktime' function. */
 #undef HAVE_MKTIME
