diff -ur Python-3.9.5/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.9.5/pyconfig.h.in	2021-05-03 10:54:42.000000000 -0400
+++ Python/pyconfig.h.in	2021-06-28 10:15:12.000000000 -0400
@@ -427,7 +427,7 @@
 #undef HAVE_GETC_UNLOCKED
 
 /* Define to 1 if you have the `getentropy' function. */
-#undef HAVE_GETENTROPY
+/* #undef HAVE_GETENTROPY */
 
 /* Define to 1 if you have the `getgrgid_r' function. */
 #undef HAVE_GETGRGID_R
