diff -ur Python-3.9.5/setup.py Python/setup.py
--- Python-3.9.5/setup.py	2021-05-03 10:54:42.000000000 -0400
+++ Python/setup.py	2021-06-25 16:50:30.000000000 -0400
@@ -2141,8 +2141,12 @@
                                sources=['_ctypes/_ctypes_test.c'],
                                libraries=['m']))
 
-        ffi_inc = sysconfig.get_config_var("LIBFFI_INCLUDEDIR")
-        ffi_lib = None
+        #ffi_inc = sysconfig.get_config_var("LIBFFI_INCLUDEDIR")
+        #ffi_lib = None
+        # Use our libffi
+        ffi_inc = sysconfig.get_config_var('INCLUDEDIR')
+        ext.library_dirs.append(sysconfig.get_config_var('LIBDIR'))
+        ffi_lib = 'ffi'
 
         ffi_inc_dirs = self.inc_dirs.copy()
         if MACOS:
