diff -ur Python-3.7.4/Lib/concurrent/futures/process.py Python/Lib/concurrent/futures/process.py
--- Python-3.7.4/Lib/concurrent/futures/process.py	2019-07-08 14:03:50.000000000 -0400
+++ Python/Lib/concurrent/futures/process.py	2019-08-13 10:30:38.000000000 -0400
@@ -457,8 +457,9 @@
             raise NotImplementedError(_system_limited)
     _system_limits_checked = True
     try:
+        # This raises FileNotFoundError on iOS
         nsems_max = os.sysconf("SC_SEM_NSEMS_MAX")
-    except (AttributeError, ValueError):
+    except (AttributeError, FileNotFoundError, ValueError):
         # sysconf not available or setting not available
         return
     if nsems_max == -1:
diff -ur Python-3.7.4/Lib/platform.py Python/Lib/platform.py
--- Python-3.7.4/Lib/platform.py	2019-07-08 14:03:50.000000000 -0400
+++ Python/Lib/platform.py	2019-08-13 10:29:44.000000000 -0400
@@ -781,6 +781,8 @@
 
     """ Interface to the system's uname command.
     """
+    # Can't open a subprocess on iOS
+    return default
     if sys.platform in ('dos', 'win32', 'win16'):
         # XXX Others too ?
         return default
@@ -804,6 +806,8 @@
         default in case the command should fail.
 
     """
+    # Can't open a subprocess on iOS
+    return default
     if sys.platform in ('dos', 'win32', 'win16'):
         # XXX Others too ?
         return default
diff -ur Python-3.7.4/Lib/urllib/request.py Python/Lib/urllib/request.py
--- Python-3.7.4/Lib/urllib/request.py	2019-07-08 14:03:50.000000000 -0400
+++ Python/Lib/urllib/request.py	2019-08-13 10:33:56.000000000 -0400
@@ -2579,7 +2579,9 @@
     return False
 
 
-if sys.platform == 'darwin':
+# _scproxy is not available on iOS
+#if sys.platform == 'darwin':
+if False:
     from _scproxy import _get_proxy_settings, _get_proxies
 
     def proxy_bypass_macosx_sysconf(host):
diff -ur Python-3.7.4/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.7.4/pyconfig.h.in	2019-07-08 14:03:50.000000000 -0400
+++ Python/pyconfig.h.in	2019-08-13 10:29:44.000000000 -0400
@@ -837,7 +837,7 @@
 #undef HAVE_SEM_UNLINK
 
 /* Define to 1 if you have the `sendfile' function. */
-#undef HAVE_SENDFILE
+/* #undef HAVE_SENDFILE */ /* sendfile seems always to generate SIGSYS on iOS */
 
 /* Define to 1 if you have the `setegid' function. */
 #undef HAVE_SETEGID
