diff -ur Python-3.12.3/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.12.3/pyconfig.h.in	2024-04-09 04:09:14
+++ Python/pyconfig.h.in	2024-05-06 15:21:36
@@ -806,13 +806,13 @@
 #undef HAVE_MKFIFO
 
 /* Define to 1 if you have the `mkfifoat' function. */
-#undef HAVE_MKFIFOAT
+/* #undef HAVE_MKFIFOAT */
 
 /* Define to 1 if you have the `mknod' function. */
 #undef HAVE_MKNOD
 
 /* Define to 1 if you have the `mknodat' function. */
-#undef HAVE_MKNODAT
+/* #undef HAVE_MKNODAT */
 
 /* Define to 1 if you have the `mktime' function. */
 #undef HAVE_MKTIME
