diff -ur Python-3.10.2/Lib/distutils/tests/test_build_ext.py Python/Lib/distutils/tests/test_build_ext.py
--- Python-3.10.2/Lib/distutils/tests/test_build_ext.py	2022-01-13 13:52:14.000000000 -0500
+++ Python/Lib/distutils/tests/test_build_ext.py	2022-02-18 11:38:14.000000000 -0500
@@ -444,6 +444,8 @@
         #  the interpreter.
         self._try_compile_deployment_target('==', None)
 
+    @unittest.skip('-mmacosx-version-min in CFLAGS overrides '
+                   'MACOSX_DEPLOYMENT_TARGET in environment')
     @unittest.skipUnless(sys.platform == 'darwin', 'test only relevant for MacOSX')
     def test_deployment_target_too_low(self):
         # Issue 9516: Test that an extension module is not allowed to be
@@ -451,6 +453,8 @@
         self.assertRaises(DistutilsPlatformError,
             self._try_compile_deployment_target, '>', '10.1')
 
+    @unittest.skip('-mmacosx-version-min in CFLAGS overrides '
+                   'MACOSX_DEPLOYMENT_TARGET in environment')
     @unittest.skipUnless(sys.platform == 'darwin', 'test only relevant for MacOSX')
     def test_deployment_target_higher_ok(self):
         # Issue 9516: Test that an extension module can be compiled with a
