diff -ur Python-3.10.2/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.10.2/pyconfig.h.in	2022-01-13 13:52:14.000000000 -0500
+++ Python/pyconfig.h.in	2022-02-18 11:46:41.000000000 -0500
@@ -442,7 +442,7 @@
 #undef HAVE_GETC_UNLOCKED
 
 /* Define to 1 if you have the `getentropy' function. */
-#undef HAVE_GETENTROPY
+/* #undef HAVE_GETENTROPY */
 
 /* Define to 1 if you have the `getgrgid_r' function. */
 #undef HAVE_GETGRGID_R
