diff -ur Python-3.7.0/pyconfig.h.in Python/pyconfig.h.in
--- Python-3.7.0/pyconfig.h.in	2018-06-26 23:07:35.000000000 -0400
+++ Python/pyconfig.h.in	2018-07-06 12:43:10.000000000 -0400
@@ -367,7 +367,7 @@
 #undef HAVE_FTRUNCATE
 
 /* Define to 1 if you have the `futimens' function. */
-#undef HAVE_FUTIMENS
+/* #undef HAVE_FUTIMENS */
 
 /* Define to 1 if you have the `futimes' function. */
 #undef HAVE_FUTIMES
@@ -1201,7 +1201,7 @@
 #undef HAVE_UTIL_H
 
 /* Define to 1 if you have the `utimensat' function. */
-#undef HAVE_UTIMENSAT
+/* #undef HAVE_UTIMENSAT */
 
 /* Define to 1 if you have the `utimes' function. */
 #undef HAVE_UTIMES
